src/components/stepper/stepper/stepper.ts

```diff
@@ -0,0 +1,278 @@
+import {
+  type CSSResultGroup,
+  html,
+  LitElement,
+  type TemplateResult,
+  type PropertyValues,
+} from 'lit';
+import { customElement, property } from 'lit/decorators.js';
+
+import { breakpoints, isBreakpoint } from '../../core/dom';
+import { ConnectedAbortController } from '../../core/eventing';
+import type { SbbHorizontalFrom, SbbOrientation } from '../../core/interfaces';
+import type { SbbStepElement, SbbStepValidateEventDetails } from '../step/step';
+
+import style from './stepper.scss?lit&inline';
+
+const DEBOUNCE_TIME = 150;
+
+/**
+ * Provides a structured, step-by-step workflow for user interactions.
+ *
+ * @slot step-label - Use this slot to provide an `sbb-step-label`.
+ * @slot step - Use this slot to provide an `sbb-step`.
+ */
+@customElement('sbb-stepper')
+export class SbbStepperElement extends LitElement {
+  public static override styles: CSSResultGroup = style;
+
+  /**
+   * If set to true, only the current and previous labels can be clicked and selected.
+   */
+  @property({ type: Boolean }) public linear = false;
+
+  /**
+   * Overrides the behaviour of `orientation` property.
+   */
+  @property({ attribute: 'horizontal-from', reflect: true })
+  public get horizontalFrom(): SbbHorizontalFrom | undefined {
+    return this._horizontalFrom;
+  }
+  public set horizontalFrom(value: SbbHorizontalFrom) {
+    this._horizontalFrom = breakpoints.includes(value) ? value : undefined;
+    if (this._horizontalFrom) {
+      this._checkOrientation();
+    }
+  }
+  private _horizontalFrom?: SbbHorizontalFrom | undefined;
+
+  /**
+   * Steps orientation, either horizontal or vertical.
+   */
+  @property({ reflect: true })
+  public orientation: SbbOrientation = 'horizontal';
+
+  /**
+   * The currently selected step.
+   */
+  public get selected(): SbbStepElement | undefined {
+    return this.querySelector<SbbStepElement>('sbb-step[data-selected]') ?? undefined;
+  }
+
+  public set selected(step: SbbStepElement) {
+    this._select(step);
+  }
+
+  /**
+   * The currently selected step index.
+   */
```

Comment:
should it not have a property annotation? or shouldn't it work when setting as attribute?